Source/WebKit/NetworkProcess/NetworkLoadScheduler.cpp

```diff
@@ -74,44 +76,44 @@ void NetworkLoadScheduler::HostContext::schedule(NetworkLoad& load)
         return;
     }
 
-    m_pendingLoads.add(&load);
+    m_pendingLoads.add(load);
 }
 
 void NetworkLoadScheduler::HostContext::unschedule(NetworkLoad& load)
 {
-    m_activeLoads.remove(&load);
-    m_pendingLoads.remove(&load);
+    m_activeLoads.remove(load);
+    m_pendingLoads.remove(load);
 
-    if (m_pendingLoads.isEmpty())
+    if (m_pendingLoads.isEmptyIgnoringNullReferences())
         return;
     if (shouldDelayLowPriority())
         return;
 
-    start(*m_pendingLoads.takeFirst());
+    start(m_pendingLoads.takeFirst());
```

Comment:
Is it inefficient to separately check for null and extract the first? Do we need an idiom that does both in a single operation to avoid doing things twice?

Reply:
Ok, I think I'll add a `tryTakeFirst()` which returns a raw pointer.